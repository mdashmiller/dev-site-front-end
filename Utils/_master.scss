$old-paper: #f0f0e3;
$lightest-grey: #bdbdbd;
$light-grey: #7e7e7e;
$med-grey: #9e9e9e;
$dark-grey: #212121;
$darker-grey: #1a1a1a;
$dark-pink: #ad1457;
$darker-pink: #880e4f;
$lighter-pink: #c2185b;
