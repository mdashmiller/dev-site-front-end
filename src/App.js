import React, { Component } from 'react'
import M from 'materialize-css'

import Header from './components/Header'
import Main from './components/Main'
import Footer from './components/Footer'
import MiniNav from './components/MiniNav'

class App extends Component {

  componentDidMount() {
    // initialize Materialize.css
    // scrollspy, tooltips and
    // parallax
    M.AutoInit()
  }

  render() {
    return (
      <div className="App">
        <Header data-test="header" />
        <Main data-test="main" />
        <MiniNav data-test="mini-nav" />
        <Footer data-test="footer" />
      </div>
    )
  }
}

export default App
