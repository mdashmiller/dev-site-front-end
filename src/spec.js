import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import App from './App'

import { findByTestAttr } from '../Utils'

// render component for testing
const setUp = (props={}) => {
  return shallow(<App {...props} />)
}

describe('App rendering', () => {

  let component
  beforeAll(() => {
    component = setUp()
  })

  it('should render 1 Header component', () => {
    const wrapper = findByTestAttr(component, 'header')
    expect(wrapper.length).toBe(1)
  })

  it('should render 1 Main component', () => {
    const wrapper = findByTestAttr(component, 'main')
    expect(wrapper.length).toBe(1)
  })

  it('should render 1 MiniNav component', () => {
    const wrapper = findByTestAttr(component, 'mini-nav')
    expect(wrapper.length).toBe(1)
  })

  it('should render 1 Footer component', () => {
    const wrapper = findByTestAttr(component, 'footer')
    expect(wrapper.length).toBe(1)
  })

})

describe('App mounting and unmounting', () => {

  it('should render without error', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

})

// describe('App snapshot', () => {

//   it('should have a valid snapshot', () => {
//     const component = renderer.create(<App />)
//     let tree = component.toJSON()
//     expect(tree).toMatchSnapshot()
//   })

// })
