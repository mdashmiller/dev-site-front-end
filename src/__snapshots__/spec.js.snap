// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should have a valid snapshot 1`] = `
<div
  className="App"
>
  <h1
    data-test="title"
  >
    My Dev Site! 2.0
  </h1>
</div>
`;
